internal/installercache/installercache.go

```diff
@@ -183,35 +282,31 @@ func (i *Installers) evict() {
 		return nil
 	}
 
-	err := filepath.Walk(i.cacheDir, visit)
+	err := filepath.Walk(i.config.CacheDir, visit)
 	if err != nil {
 		if !os.IsNotExist(err) { //ignore first invocation where the cacheDir does not exist
-			i.log.WithError(err).Errorf("release binary eviction failed to inspect directory %s", i.cacheDir)
+			i.log.WithError(err).Errorf("release binary eviction failed to inspect directory %s", i.config.CacheDir)
 		}
-		return
+		return false
 	}
 
-	//prune the hard links just in case the deletion of resources
-	//in ignition.go did not succeeded as expected
-	for idx := 0; idx < len(links); idx++ {
-		finfo := links[idx]
-		//Allow a grace period of 5 minutes from the link creation time
-		//to ensure the link is not being used.
-		grace := time.Now().Add(DeleteGracePeriod).Unix()
-		if finfo.info.ModTime().Unix() < grace {
-			os.Remove(finfo.path)
-		}
-	}
+	// TODO: We might want to consider if we need to do this longer term, in theory every hardlink should be automatically freed.
+	i.pruneExpiredHardLinks(links, linkPruningGracePeriod)
 
-	//delete the oldest file if necessary
-	for totalSize >= int64(float64(i.storageCapacity)*CacheLimitThreshold) {
-		finfo, _ := files.Pop()
+	// delete the oldest file if necessary
+	evicted := false
+	q := i.filterOutHardlinkedReleases(files)
```

Comment:
Since we now only use the one list (files without hardlinks) can we do this filtering in the `visit` function rather than in this post-processing?

Reply:
I considered that, the only thing that worries me is that I do have a piece of code in this filtering method that takes some action if a stat fails. 

In initially preferred to avoid this just in case I introduce a problem, but I can take another look. 

I can see the benefit of cutting down on a few cycles